### PR TITLE
[IRGen] Add getEnumTag and injectEnumTag builtins

### PR DESCRIPTION
**Pull request by Azoy**
This adds 2 new builtins, one that calls the getEnumTag vwt function on a generic type and another that injects the given tag into some generic value. These are unsafe operations that assume the value passed in as `T` is an enum type.

### CODE REVIEW

File: lib/AST/Builtins.cpp [hunk 1]
```diff
@@ -1968,6 +1968,26 @@ static ValueDecl *getPackLength(ASTContext &ctx, Identifier id) {
   return builder.build(id);
 }
 
+static ValueDecl *getGetEnumTag(ASTContext &ctx, Identifier id) {
+  BuiltinFunctionBuilder builder(ctx, /* genericParamCount */ 1);
```
Comment (nate-chandler) [inline lib/AST/Builtins.cpp:1972]: Would be nice to use the getBuiltinFunction infrastructure here.

**Review by nate-chandler — APPROVED** [review]
lgtm!

**Comment by Azoy** [comment]
@swift-ci please test

**Comment by Azoy** [comment]
@swift-ci please test Linux

**Pull request merged**